File: src/main/java/com/al3x/housing2/Enums/ParticleType.java
package com.al3x.housing2.Enums;

import org.bukkit.Material;

public enum ParticleType implements EnumMaterial{
    LINE(Material.STICK),
    CIRCLE(Material.SNOWBALL),
    SQUARE(Material.PAINTING);

    Material material;

    ParticleType(Material material) {
        this.material = material;
    }

    @Override
    public Material getMaterial() {
        return material;
    }
}

File: paper/src/main/java/com/al3x/housing2/Enums/ParticleType.kt
package com.al3x.housing2.Enums;

import org.bukkit.Material;

enum class ParticleType(override val material: Material) : EnumMaterial {
    LINE(Material.STICK),
    CIRCLE(Material.SNOWBALL),
    CURVE(Material.EGG),
    SQUARE(Material.PAINTING);
}
